Spokestack/SpeechContext.swift

```diff
@@ -56,40 +56,58 @@ import Foundation
+        switch event {
+        case .initialize:
+            self.listeners.forEach { listener in
+                self.configuration.delegateDispatchQueue.async {
                     listener.didInit()
-                case .start:
+                }}
+        case .start:
+            self.listeners.forEach { listener in
+                self.configuration.delegateDispatchQueue.async {
                     listener.didStart()
-                case .stop:
+                }}
+        case .stop:
+            self.listeners.forEach { listener in
+                self.configuration.delegateDispatchQueue.async {
                     listener.didStop()
-                case .activate:
+                }}
+        case .activate:
+            self.listeners.forEach { listener in
+                self.configuration.delegateDispatchQueue.async {
                     listener.didActivate()
-                case .deactivate:
+                }}
+        case .deactivate:
+            self.listeners.forEach { listener in
+                self.configuration.delegateDispatchQueue.async {
                     listener.didDeactivate()
-                case .recognize:
+                }}
+        case .recognize:
+            self.listeners.forEach { listener in
+                self.configuration.delegateDispatchQueue.async {
                     listener.didRecognize(self)
-                case .error:
-                    guard let e = self.error else {
-                        listener.failure(speechError: SpeechPipelineError.errorNotSet("A pipeline component attempted to send an error to SpeechContext's listeners without first setting the SpeechContext.error property."))
-                        return
-                    }
+                }}
+        case .error:
+            let e = (self.error != nil) ? self.error! : SpeechPipelineError.errorNotSet("A pipeline component attempted to send an error to SpeechContext's listeners without first setting the SpeechContext.error property.")
+            self.listeners.forEach { listener in
+                self.configuration.delegateDispatchQueue.async {
                     listener.failure(speechError: e)
-                    self.error = .none
-                case .trace:
-                    guard let t = self.trace else {
-                        // swallow this error case because traces aren't important to the client.
-                        return
-                    }
-                    listener.didTrace(t)
-                    self.trace = .none
-                case .timeout:
-                    listener.didTimeout()
-                }
+                }}
+            self.error = .none
+        case .trace:
+            if let t = self.trace {
+                self.listeners.forEach { listener in
+                    self.configuration.delegateDispatchQueue.async {
+                        listener.didTrace(t)
+                    }}
+                self.trace = .none
```

Comment:
If you really want to clear this message out (something that's not done in Android, and it actually just helped me catch a bug, almost by accident), you could still save yourself some code by moving the switch back where you had it and checking for the trace event after the loop's over, clearing it there.